tests/utils/test_distributed_utils.py

```diff
@@ -0,0 +1,851 @@
+
+class TestClientFunctions:
+    @pytest.mark.gpu
+    @patch("nemo_curator.utils.distributed_utils.LocalCUDACluster")
+    @patch("nemo_curator.utils.distributed_utils.Client")
+    @patch("nemo_curator.utils.distributed_utils._set_torch_to_use_rmm")
+    def test_start_dask_gpu_local_cluster(
+        self, mock_set_torch_rmm, mock_client, mock_cuda_cluster
+    ):
+        """Test starting a GPU local cluster."""
+        # Setup mock return values
+        mock_cluster_instance = MagicMock()
+        mock_cuda_cluster.return_value = mock_cluster_instance
+        mock_client_instance = MagicMock()
+        mock_client.return_value = mock_client_instance
+
+        # Mock get_num_workers to return a positive value
+        with patch(
+            "nemo_curator.utils.distributed_utils.get_num_workers", return_value=1
+        ):
+            # Call the function
+            client = start_dask_gpu_local_cluster()
+
+            # Verify the client was created with the right cluster
+            mock_client.assert_called_once_with(mock_cluster_instance)
+            assert client == mock_client_instance
+
+            # Verify _set_torch_to_use_rmm was called
+            mock_set_torch_rmm.assert_called_once()
+
+    @patch("nemo_curator.utils.distributed_utils.LocalCluster")
+    @patch("nemo_curator.utils.distributed_utils.Client")
+    def test_start_dask_cpu_local_cluster(self, mock_client, mock_local_cluster):
+        """Test starting a CPU local cluster."""
+        # Setup mock return values
+        mock_cluster_instance = MagicMock()
+        mock_local_cluster.return_value = mock_cluster_instance
+        mock_client_instance = MagicMock()
+        mock_client.return_value = mock_client_instance
+
+        # Mock get_num_workers to return a positive value
+        with patch(
+            "nemo_curator.utils.distributed_utils.get_num_workers", return_value=1
+        ):
+            # Call the function
+            client = start_dask_cpu_local_cluster()
+
+            # Verify the client was created with the right cluster
+            mock_client.assert_called_once_with(mock_cluster_instance)
+            assert client == mock_client_instance
+
+    @patch("nemo_curator.utils.distributed_utils.start_dask_gpu_local_cluster")
+    @patch("nemo_curator.utils.distributed_utils.start_dask_cpu_local_cluster")
+    @patch("nemo_curator.utils.distributed_utils.Client")
+    def test_get_client(self, mock_client, mock_cpu_cluster, mock_gpu_cluster):
+        """Test get_client function with different params."""
+        # Test with scheduler_address
+        mock_client_instance = MagicMock()
+        mock_client.return_value = mock_client_instance
+
+        # Mock get_num_workers to return a positive value
+        with patch(
+            "nemo_curator.utils.distributed_utils.get_num_workers", return_value=1
+        ):
+            # Test with scheduler_address
+            client = get_client(scheduler_address="tcp://localhost:8786")
```

Comment:
Add case when both `scheduler_address` and `scheduler_file` are specified?

Reply:
Check `assert get_num_workers(client) > 0, "No workers are currently connected."`?